contracts/Example2.sol

```diff
@@ -1,11 +1,26 @@
+// SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.4;
 
 contract Example2 {
     uint public counter;
- 
-    function incrementBy(uint[] memory arr) public {
-        for (uint idx = 0; idx < arr.length; idx++){
-            counter += arr[idx];
+
+    function incrementBy(uint[] calldata arr) external
+    {
+        uint _counter;
+        uint idx;
+        uint _length = arr.length;
+
+        // Not a good idea in general to wrap the loop in unchecked, but it works for this case
+        // and we squeeze another 1000 gas when compared to the original for loop
+        unchecked
+        {
+            while (idx != _length)
+            {
+                _counter += arr[idx]; // 'OK' to overflow
```

Comment:
Why is it okay to overflow the counter? The unchecked should only be for `++idx` because we know it will anyhow not overflow due to the check with length

Reply:
I thought of it as "doesn't matter if it overflows", and wanted to try out the unchecked over a while loop
but defo not a good idea